Source/Core/VideoCommon/PixelShaderManager.cpp

```diff
@@ -110,15 +110,20 @@ void PixelShaderManager::SetConstants()
       // they are the coefficients from the center to the border of the screen
       // so to simplify I use the hi coefficient as K in the shader taking 256 as the scale
       // TODO: Shouldn't this be EFBToScaledXf?
-      constants.fogf[0][0] = ScreenSpaceCenter;
-      constants.fogf[0][1] =
+      constants.fogf[0][2] = ScreenSpaceCenter;
+      constants.fogf[0][3] =
           static_cast<float>(g_renderer->EFBToScaledX(static_cast<int>(2.0f * xfmem.viewport.wd)));
-      constants.fogf[0][2] = bpmem.fogRange.K[4].HI / 256.0f;
+
+      for (size_t i = 0; i < ArraySize(bpmem.fogRange.K); i++)
+      {
+        constants.fogf[i][0] = bpmem.fogRange.K[i].GetValue(0);
+        constants.fogf[i][1] = bpmem.fogRange.K[i].GetValue(1);
+      }
     }
     else
     {
-      constants.fogf[0][0] = 0;
-      constants.fogf[0][1] = 1;
+      constants.fogf[0][2] = 0;
+      constants.fogf[0][3] = 1;
       constants.fogf[0][2] = 1;
```

Comment:
Did you actually want to set `constants.fogf[0][2]` twice here? This looks like an oversight to me.

Reply:
Nope. Thanks, that is indeed redundant.